Source/WebCore/platform/graphics/skia/FontCascadeSkia.cpp

```diff
@@ -30,74 +30,17 @@
 #include "FontCache.h"
 #include "GraphicsContextSkia.h"
 #include "SurrogatePairAwareTextIterator.h"
-
-WTF_IGNORE_WARNINGS_IN_THIRD_PARTY_CODE_BEGIN
-#include <skia/core/SkTextBlob.h>
-WTF_IGNORE_WARNINGS_IN_THIRD_PARTY_CODE_END
 #include <wtf/text/CharacterProperties.h>
 
 namespace WebCore {
 
-WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN // GLib/Win port
-
 void FontCascade::drawGlyphs(GraphicsContext& graphicsContext, const Font& font, std::span<const GlyphBufferGlyph> glyphs, std::span<const GlyphBufferAdvance> advances, const FloatPoint& position, FontSmoothingMode smoothingMode)
 {
-    if (!font.platformData().size())
-        return;
-
-    const auto& fontPlatformData = font.platformData();
-    const auto& skFont = fontPlatformData.skFont();
-
-    if (!font.allowsAntialiasing())
-        smoothingMode = FontSmoothingMode::NoSmoothing;
-
-    SkFont::Edging edging;
-    switch (smoothingMode) {
-    case FontSmoothingMode::AutoSmoothing:
-        edging = skFont.getEdging();
-        break;
-    case FontSmoothingMode::Antialiased:
-        edging = SkFont::Edging::kAntiAlias;
-        break;
-    case FontSmoothingMode::SubpixelAntialiased:
-        edging = SkFont::Edging::kSubpixelAntiAlias;
-        break;
-    case FontSmoothingMode::NoSmoothing:
-        edging = SkFont::Edging::kAlias;
-        break;
-    }
-
-    bool isVertical = fontPlatformData.orientation() == FontOrientation::Vertical;
-    SkTextBlobBuilder builder;
-    const auto& buffer = [&]() {
-        if (skFont.getEdging() == edging)
-            return isVertical ? builder.allocRunPos(skFont, glyphs.size()) : builder.allocRunPosH(skFont, glyphs.size(), 0);
-
-        SkFont copiedFont = skFont;
-        copiedFont.setEdging(edging);
-        return isVertical ? builder.allocRunPos(copiedFont, glyphs.size()) : builder.allocRunPosH(copiedFont, glyphs.size(), 0);
-    }();
-
-    FloatSize glyphPosition;
-    for (size_t i = 0; i < glyphs.size(); ++i) {
-        buffer.glyphs[i] = glyphs[i];
-
-        if (isVertical) {
-            glyphPosition += advances[i];
-            buffer.pos[2 * i] = glyphPosition.height();
-            buffer.pos[2 * i + 1] = glyphPosition.width();
-        } else {
-            buffer.pos[i] = glyphPosition.width();
-            glyphPosition += advances[i];
-        }
-    }
-
-    auto blob = builder.make();
-    static_cast<GraphicsContextSkia*>(&graphicsContext)->drawSkiaText(blob, SkFloatToScalar(position.x()), SkFloatToScalar(position.y()), edging != SkFont::Edging::kAlias, isVertical);
+    bool enableAntialiasing, isVertical;
```

Comment:
We only ever use one variable per line. And I'd strongly advocate to initialize them with `{ false }` - to not depend on whatever `buildTextBlob` does.